hedera-node/src/main/java/com/hedera/services/context/ServicesContext.java

```diff
@@ -612,20 +605,57 @@ public ServicesContext(
 		this.platform = platform;
 		this.state = state;
 		this.propertySources = propertySources;
+
+		updateWorkingState(state);
+		updateQueryableState(state);
 	}
 
+	/**
+	 * Update the state and working state based on the provided service state
+	 *
+	 * @param state latest state from the services
+	 */
 	public void update(ServicesState state) {
 		this.state = state;
 
-		queryableAccounts().set(accounts());
-		queryableTopics().set(topics());
-		queryableStorage().set(storage());
-		queryableTokens().set(tokens());
-		queryableTokenAssociations().set(tokenAssociations());
-		queryableSchedules().set(schedules());
-		queryableUniqueTokens().set(uniqueTokens());
-		queryableUniqueTokenAssociations().set(uniqueTokenAssociations());
-		queryableUniqueOwnershipAssociations().set(uniqueOwnershipAssociations());
+		updateWorkingState(state);
+		updateQueryableState(state);
+	}
+
+	/**
+	 * Update the queryable state
+	 */
+	private void updateQueryableState(ServicesState state) {
```

Comment:
Suppose a query thread uses both the `accounts` and `tokens` maps. 

With this update pattern, the query thread could get the `accounts` map from the "old" state, and then the `tokens` map from the "new" state. (That is, there is a race between the thread updating the queryable state children one-by-one, and the query thread _accessing_ the children.)

To make sure the `StateChildren` returned by `queryableState.get()` has a consistent view of all maps, we can instead do:
```
final var newQueryableStateChildren = new StateChildren();
newQueryableStateChildren.setAccounts(state.accounts());
...
newQueryableStateChildren.setUniqueOwnershipAssociations(state.uniqueOwnershipAssociations());
queryableState.set(newQueryableStateChildren);
```

Reply:
Excellent point, made this change. Thanks!